docs/release-notes.txt

```diff
@@ -6,6 +6,39 @@ Release Notes
 Version 0.12
 ------------
 
+Version 0.12.7
+^^^^^^^^^^^^^^
+
+**Release Date:** June 11, 2020
+
+- **Breaking Change:** Gradient clipping for PyTorchTrial should now be specified via `determined.PyTorch.PyTorchCallback` via the `on_before_optimizer_step()` instead of being specified via the experiment configuration. Determined provides two pre-made callbacks for gradient clipping:  `determined.PyTorch.ClipGradsL2Norm` and `determined.PyTorch.ClipGradsL2Value`.
+
+- Add support for Keras callbacks that stop training by setting the model.stop_training flag to True, including the official EarlyStopping callback: https://www.tensorflow.org/api_docs/python/tf/keras/callbacks/EarlyStopping. When a stop is requested, we will finish the training (or validation) step we are in, checkpoint, and terminate the trial.
+
+- Add support for Estimator hooks that stop training by calling run_context.request_stop(), including the official stop_if_no_decrease_hook: https://www.tensorflow.org/api_docs/python/tf/estimator/experimental/stop_if_no_decrease_hook. When a stop is requested, we will finish the training (or validation) step we are in, checkpoint, and terminate the trial.
+
+- Add support for requesting training to stop programmatically from trial code. We recommend using the official Keras callbacks or Estimator hooks if you are using those frameworks. For PyTorch, you can request a stoppage using the context by calling context.set_stop_requested(True) from a PyTorch callback. When a stop is requested, we will finish the training step we are in, checkpoint, and terminate the trial.
+
+- Support errored trials in hyperparameter searches. Before if an error was hit, the whole experiment and search would exit. This also works with trials that use an early stopping callback.
+
+- Support multi-GPU training for TF2 models that use IndexedSlices for model parameters.
```

Comment:
blocking: "TensorFlow 2" and monospace IndexedSlices

Reply:
hopefully it's clear that the line I'm referring to is always the bottom line in the snippet lol